openstack/clientconfig/requests.go

```diff
@@ -745,7 +745,11 @@ func NewServiceClient(service string, opts *ClientOpts) (*gophercloud.ServiceCli
 	}
 
 	// Get a Provider Client
-	pClient, err := AuthenticatedClient(opts)
+	ao, err := AuthOptions(opts)
+	if err != nil {
+		return nil, err
+	}
+	pClient, err := openstack.NewClient(ao.IdentityEndpoint)
```

Comment:
How about we take the opportunity to move the creation of `pClient` closer to where it's used, on line 803?

Reply:
@mandre fixed with 9cc2ea9